### ISSUE TITLE
Router bewertet nur den ersten Token des Tool-Namens — HA-Toolset kollabiert auf einen Score

### ISSUE THREAD

**Issue by geisten** [geisten/geistlib#140]
## Befund

`agent_score_names` bewertet jede Option mit `agent_first_token_logit`, und die liefert `logits[ids[0]]` — den Logit **nur des ersten Tokens** des Tool-Namens. Zwei Tools, deren Namen auf denselben ersten Token tokenisieren, bekommen damit einen **identischen Score** und sind für den Router nicht unterscheidbar.

Gemessen (Llama-3.2-Tokenizer, `geist_session_tokenize`):

```
TurnOn               erster Token=19952  "Turn"
TurnOff              erster Token=19952  "Turn"     <- identisch
GetState             erster Token=1991   "Get"
SetBrightness        erster Token=1681   "Set"

HassTurnOn           erster Token=39     "H"
HassTurnOff          erster Token=39     "H"
HassGetState         erster Token=39     "H"
HassSetBrightness    erster Token=39     "H"        <- alle vier identisch
```

**Für das Home-Assistant-Produkt kollabiert damit das gesamte Toolset auf einen einzigen Score.** Jedes Tool trägt das `Hass`-Präfix, also kann erzwungenes Routing dort grundsätzlich keines seiner Werkzeuge auswählen. Siehe geisten/geist-home-assistant#39.

## Warum Umbenennen keine verlässliche Abhilfe ist

Naheliegend wäre, Tools so zu benennen, dass die ersten Token differieren. Gemessen (40 Fälle, forced routing, margin 0.0), Namen `TurnOn/TurnOff/GetState/SetBrightness` → `Activate/Deactivate/Query/Dim`:

| Modell | Originalnamen | unterscheidbare Namen |
|---|---|---|
| Gemma 4 E2B | 0.700 | **0.833** |
| BitNet b1.58 2B-4T | 0.476 | 0.381 |
| Llama 3.2 1B | 0.381 | 0.286 |

Gemma gewinnt deutlich, die beiden anderen verlieren. Der Grund liegt nahe: Umbenennen tauscht **Token-Unterscheidbarkeit gegen lexikalische Passung**. Die Anfrage sagt „Turn on the kitchen light" — `TurnOn` passt wörtlich, `Activate` nicht. Wo die Passung mehr wog als die Kollision, verschlechtert die Umbenennung das Ergebnis.

Ein Host kann das Problem also nicht sauber umgehen: Er müsste Namen wählen, die gleichzeitig lexikalisch passen *und* im ersten Token differieren — was bei Synonympaaren wie an/aus prinzipiell schwer ist.

## Erwartung

Die Bewertung sollte mehr als den ersten Token berücksichtigen — etwa die Summe oder das Mittel der Logits über die Tokenfolge des Namens, oder eine Bewertung der vollständigen Namenssequenz. Das gäbe Unterscheidbarkeit, **ohne** die lexikalische Passung zu opfern.

Nebenbefund derselben Messreihe: Reichere Tool-Beschreibungen wirken uneinheitlich (Kontrollmodell 0.381 → 0.571, BitNet 0.476 → 0.429), was zur Analyse passt — die Entscheidung hängt am Namen, nicht an der Beschreibung.

Harness, Rohdaten und Reproduktion: geisten/geistagent (`docs/CALIBRATION.md`, `eval/results/`).
